.github/workflows/release-tgz.yml

```diff
@@ -0,0 +1,20 @@
+name: Release
+on: workflow_call
+permissions:
+  contents: write
+jobs:
+  upload:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Determine version from tag name
+        id: vars
+        run: echo version="${tag_name#v}" >> $GITHUB_OUTPUT
+        env:
+          tag_name: ${{github.event.release.tag_name}}
+      - name: Package sources into tar.gz
+        run: git ls-tree -r -z --name-only HEAD | tar --null --files-from=- --transform="flags=r;s:^:${{github.event.repository.name}}-${{steps.vars.outputs.version}}/:" --sort=name --mtime=2030-01-01T00:00:00Z --owner=0 --group=0 --numeric-owner --create --gzip --file=${{github.event.repository.name}}-${{steps.vars.outputs.version}}.tar.gz
```

Comment:
why not just `git archive` ?

Reply:
git archive doesn't have  `--sort` or `--owner` options